src/vs/workbench/common/editor/editorGroupModel.ts

```diff
@@ -373,16 +361,27 @@ export class EditorGroupModel extends Disposable {
 		this.splice(replaceIndex, false, replaceWith);
 
 		if (event) {
-			this._onDidCloseEditor.fire(event);
+			this._onDidModelChange.fire({
+				kind: GroupChangeKind.EDITOR_CLOSE,
+				editor: event.editor,
+				editorIndex: event.index,
+				closeContext: event.context,
+				closedSticky: event.sticky
+			});
 		}
 	}
 
 	closeEditor(candidate: EditorInput, context = EditorCloseContext.UNKNOWN, openNext = true): IEditorCloseEvent | undefined {
 		const event = this.doCloseEditor(candidate, context, openNext);
```

Comment:
This is now a bit backwards, why would `doCloseEditor` return an `event` when it is not used?

Reply:
What do you mean when it's not used? I just replaced the firing call everything else is the same

Reply:
Yeah previously the event was literally fired and now we morph it into the new event, we could rename the return type to be `ICloseEditorResult` or similar.